lib/puppet/defaults.rb

```diff
@@ -1689,6 +1689,31 @@ def self.initialize_default_settings!(settings)
         new configurations, where you want to fix the broken configuration
         rather than reverting to a known-good one.",
     },
+    :factnamelength_soft_limits => {
```

Comment:
Do we have a preference for `factnamelength_soft_limits` over `fact_name_length_soft_limits`? Same for the other settings.
I find the second one easier to read.

Reply:
Also, will there ever be a `fact_name_length_hard_limit`? If not, I'd find names in the vein of `max_fact_name_length` more readable (same as what we did with `max_files`).

Reply:
It is possible that hard limits will exist in the future. They are not yet implemented in this PR since we don't have a good idea on how these limits might look like, and it's better to push out the soft limits first and get some feedback on those, so we can later decide what the hard limits will be. Otherwise we risk stoping agent runs due to badly chosen limits.